pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -508,6 +508,8 @@ def args_as_dict(self) -> dict[str, Any]:
         """
         if isinstance(self.args, dict):
             return self.args
+        if isinstance(self.args, str) and not self.args:
+            return {}
```

Comment:
```suggestion
        if not self.args:
            return {}
```

Can we simply check this?

Reply:
The issue with this suggestion is that it'll consider `None`, `False`, `''`, 0  as valid values, and that could lead to silently letting slip by some incorrect values. Checking that the args is a string makes it clear what type is allowed